warehouse/router.go

```diff
@@ -691,12 +713,40 @@ func (r *router) createUploadJobsFromStagingFiles(ctx context.Context, warehouse
 
 	// reset upload trigger if the upload was triggered
 	if uploadTriggered {
-		clearTriggeredUpload(warehouse)
+		r.triggerStore.ClearTrigger(warehouse.Identifier)
 	}
 
 	return nil
 }
 
+func (r *router) uploadFrequencyExceeded(warehouse model.Warehouse, syncFrequency string) bool {
+	freqInS := r.uploadFreqInS(syncFrequency)
+
+	r.createJobMarkerMapLock.RLock()
+	defer r.createJobMarkerMapLock.RUnlock()
+
+	lastCreatedAt, ok := r.createJobMarkerMap[warehouse.Identifier]
+	if !ok {
+		return true
+	}
+	return r.now().Sub(lastCreatedAt) > time.Duration(freqInS)*time.Second
```

Comment:
[very minor] it's simple enough in this case to avoid keeping the lock while calculating this:

```go
r.createJobMarkerMapLock.RLock()
lastCreatedAt, ok := r.createJobMarkerMap[warehouse.Identifier]
r.createJobMarkerMapLock.RUnlock()
```